opentracing-kotlin/src/main/kotlin/org/zalando/opentracing/kotlin/TracerExtension.kt

```diff
@@ -0,0 +1,90 @@
+package org.zalando.opentracing.kotlin
+
+import io.opentracing.Scope
+import io.opentracing.Span
+import io.opentracing.SpanContext
+import io.opentracing.Tracer
+import io.opentracing.propagation.Format
+import io.opentracing.propagation.TextMapAdapter
+import io.opentracing.tag.Tags
+
+inline fun <T : Tracer, R> T.trace(span: Span, block: (Span) -> R): R {
+
+    var scope: Scope? = null
+    var closed = false
+
+    try {
+        scope = scopeManager().activate(span)
```

Comment:
Starting/closing and activating a span are two completely separate concerns. I don't believe it's a good idea to mix them.

Reply:
I disagree: you always want to do both together - except for cases where you're in an interceptor that has a pre- and post-callback (e.g. soap).
This library is not intended for those special cases - you can still use the plain tracer.

Reply:
Nope. Scope activation only works in thread-based programming models. So, you definitely don't want to use it in reactive streams environments.

Reply:
Also, the lifecycle of a span might be bigger than the scope activation. There is no good reason to tie them together. Maybe the span is being owned and managed by someone else, but you need to propagate it using scopes. I had the need to that multiple times.

Reply:
So, the activation surely does not work on reactive streams, but then again this method is not meant for that case. These methods exist only to avoid boiler plate code from the most basic use cases of opentracing. I think abstracting the complete tracing flow where possible is the main objective of these extension functions

Reply:
1. Yes, there are exceptions - then don't use this method
2. reactive: In case of Kotlin, the preferred mechanism is coroutines - and there is a way to address this [transparent to the user](https://github.com/Shinigami072/OpenTracing-Kotlin-Coroutine-Integration/blob/daee6ed817c68a9edb25dff7b1c9459467ddcbf7/coroutine-tracing-api-core/src/main/kotlin/ActiveSpan.kt#L37-L58) - this is just currently not in focus of this PR